runtime/drivers/duckdb/olap_test.go

```diff
@@ -94,11 +96,18 @@ func TestCancel(t *testing.T) {
 				cctx, cancel := context.WithCancel(ctx)
 				ctx = cctx
 				go func() {
-					time.Sleep(time.Millisecond)
+					// ensure the queue fills up before cancelling the context
+					time.Sleep(100 * time.Millisecond)
 					cancel()
+					cancelCh <- struct{}{}
 				}()
 			}
 
+			if priority == cancelIdx {
+				// wait until context is cancelled
+				<-cancelCh
+			}
+
 			rows, err := olap.Execute(ctx, &drivers.Statement{
```

Comment:
By the time `Execute` is called, the `ctx` is already cancelled, so it's discarded before getting added to the queue. So doesn't really test that cancelled queries are removed from queue. 

I don't know if there's an easy way to test for it without enabling lower-level injection. Probably not worth it – we already have tests for cancellation for the semaphore in `priorityqueue`.

Merging now, but probably we should just delete this test case when simplifying the DuckDB tests.